### ISSUE TITLE
Check for mips64 alignment errors on hardware

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#104]
The fix in r24914 from Dusan eliminated some mips64 alignment errors once the unaligned double values had been accessed with atomic-ops, as was introduced in r24737.

Before that that change I am quite sure that those values would have been accessed from C code with ld/sd instructions [1]. Those instructions require 8-byte alignment. Since we did not have that prior to Dusan's fix, I suspect that we were getting alignment fixups from the kernel.

On our 32-bit boards we use a program called `smwrapper` [2] to disable alignment fixups, so that they will cause bus errors, forcing a failure, and making us go find and fix the source of alignment error.

Since all that did not happen with this bug, I suspect we are not running `smwrapper` or equivalent, and I expect that the kernels we use on the mips64 test boards are fixing up alignment.

Please verify if that is true, and ensure that alignment fixuips are **not** being done on our test boards, of all sizes, endian-ness, and other flavors.

[1] I had used explicit unaligned accessors in generated code, since this was flagged on the simulator. It's obvious to me now that implied access from C code that would be problematic, but I missed it at the time.

[2] I have no idea what `smwrapper` stands for. Please change the name to something meaningful. (I have no problem with "wrapper", it is the meaning of "sm" which eludes me. Please come up with a sensible name and change to that in all usages on the buildbots.)

**Comment by paul99** [comment]
@MIPSbkirby, can you please look into this? 

Also, I don't see evidence of `smwrapper` being run on the buildbots currently, even for 32-bit boards. If it is there, can you tell me how to see it? (I would have thought it would be a test-runner command prefix.)

**Comment by MIPSbkirby** [comment]
The `smwrapper` is no longer being run.  It was removed when we switched to recipes.  I've dredged up how it was run before and it should be possible to add it back if necessary.

**Comment by paul99** [comment]
I would like to add it back. Since it has not been run in a while, maybe we should either (a) manually test with it on all targets first, or (b) add it back incrementally, a few targets at a time.

It would also be great of you could rename it, or at least describe what it means? It certainly does not convey its purpose (disable kernel alignment fixups, so that alignment errors become fatal) to me.

**Comment by MIPSbkirby** [comment]
I've added the disable-kernel-alignment-fixups program to the Buildbots.  run-tests.py runs under this program disabling the kernel alignment fixups.  When a program with unaligned data runs under this program, it will generate a bus error.  I've tested all the test suites and this doesn't add any new failures.  

**Note:** disable-kernel-alignment-fixups only disables the kernel fixups only.  On Cavium processors like the Edge Router Pro or Itus boards, the processor does it's own fixups.  So, no bus errors will be generated even when running under this program.  To avoid confusion, the program checks /proc/cpuinfo to see if the CPU is a Cavium board and reports that fixups will still occur.
